programs/ziti-edge-tunnel/include/model/dtos.h

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2019 Netfoundry, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#ifndef ZITI_TUNNEL_SDK_C_DTOS_H
+#define ZITI_TUNNEL_SDK_C_DTOS_H
+
+#include "ziti/ziti.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define TUNNEL_CONFIG(XX, ...) \
+XX(ZtAPI, string, none, ZtAPI, __VA_ARGS__) \
+XX(ConfigTypes, string, array, ConfigTypes, __VA_ARGS__)
+
+#define TUNNEL_METRICS(XX, ...) \
+XX(Up, string, none, Up, __VA_ARGS__) \
+XX(Down, string, none, Down, __VA_ARGS__)
+
+#define TUNNEL_IDENTITY(XX, ...) \
```

Comment:
what is the purpose of this type? the fields are never populated

Reply:
I have used this field to store the identity instance, so we could use the instance to send the events. The changes are  in the next PR https://github.com/openziti/ziti-tunnel-sdk-c/pull/200/files